mobx_codegen/lib/src/store_class_visitor.dart

```diff
@@ -182,24 +191,24 @@ class StoreClassVisitor extends SimpleElementVisitor {
         return;
       }
 
-      if (element.isAsynchronous) {
+      if (element.fragments.any((fragment) => fragment.isAsynchronous)) {
```

Comment:
A `MethodElement2` can have multiple fragments (see [the migration guide](https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/doc/element_model_migration_guide.md) and [the augmentation libraries feature spec](https://github.com/dart-lang/language/blob/main/working/augmentation-libraries/feature-specification.md)). This can happen in the following scenario:

```dart
class MyClass {
  // This is the first fragment:
  Future<void> myMethod(int myArg);
}

// ...

augment class MyClass {
  // And this is the second fragment:
  augment myMethod(int myArg) async {
    // body here...
  }
}
```

Note that in this case only one of the fragments would actually be considered asynchronous - there is no `isAsyncrhonous` property on `MethodElement2`.